include/loader/genomicsdb_importer.h

```diff
@@ -0,0 +1,118 @@
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS 
+ * FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR 
+ * COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER 
+ * IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN 
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+*/
+#ifndef GENOMICSDB_IMPORTER_H
+#define GENOMICSDB_IMPORTER_H
+
+#include "tiledb_loader.h"
+
+class GenomicsDBImporterException : public std::exception {
+  public:
+    GenomicsDBImporterException(const std::string m="") : msg_("GenomicsDBImporterException : "+m) { ; }
+    ~GenomicsDBImporterException() { ; }
+    // ACCESSORS
+    const char* what() const noexcept { return msg_.c_str(); }
+  private:
+    std::string msg_;
+};
+
+class GenomicsDBImporter
+{
+  public:
+    GenomicsDBImporter(const std::string& loader_config_file, const int rank,
+        const int64_t lb_callset_row_idx=0, const int64_t ub_callset_row_idx=INT64_MAX-1)
+    {
+      m_is_loader_setup = false;
+      m_rank = rank;
+      m_loader_config_file = loader_config_file;
+      m_lb_callset_row_idx = lb_callset_row_idx;
+      m_ub_callset_row_idx = ub_callset_row_idx;
+      m_loader_ptr = 0;
+      m_read_state = 0;
+    }
+    //Delete copy constructor
+    GenomicsDBImporter(const GenomicsDBImporter& other) = delete;
+    //Define move constructor
+    GenomicsDBImporter(GenomicsDBImporter&& other);
+    //Destructor
+    ~GenomicsDBImporter();
+    //Buffer streams
+    void add_buffer_stream(const std::string& name, const VidFileTypeEnum buffer_stream_type,
+        const size_t capacity)
+    {
+      add_buffer_stream(name, buffer_stream_type, capacity, 0, 0);
+    }
+    void add_buffer_stream(const std::string& name, const VidFileTypeEnum buffer_stream_type,
+        const size_t capacity, const uint8_t* initialization_buffer, const size_t num_bytes_in_initialization_buffer);
+    /*
+     * No more buffer streams can be added to this object after setup_loader is called
+     */
+    void setup_loader(const std::string& buffer_stream_callset_mapping_json_string="");
+    inline const std::vector<int64_t>& get_buffer_stream_idx_to_global_file_idx_vec() const
+    {
+      assert(m_is_loader_setup);
+      return m_loader_ptr->get_buffer_stream_idx_to_global_file_idx_vec();
+    }
+    /*
+     * Can be used by callers to pre-allocate vector<BufferStreamIdentifier>
+     */
+    size_t get_max_num_buffer_stream_identifiers() const { return m_loader_ptr->get_max_num_buffer_stream_identifiers(); }
+    /*
+     * Write to buffer stream
+     */
+    void write_data_to_buffer_stream(const int64_t buffer_stream_idx, const unsigned partition_idx, const uint8_t* data, const size_t num_bytes)
```

Comment:
We should be consistent on whether to define methods inside header file or not. Mixing this makes it difficult to maintain.